amazon-kinesis-client/src/main/java/software/amazon/kinesis/leases/LeaseRefresher.java

```diff
@@ -217,6 +217,19 @@ default void updateLeaseWithMetaInfo(Lease lease, UpdateField updateField)
      */
     boolean isLeaseTableEmpty() throws DependencyException, InvalidStateException, ProvisionedThroughputException;
 
+    /**
+     * Check (synchronously) if there are any leases in the lease table for a given stream identifier.
+     *
+     * @param streamIdentifier for multi-stream mode. Can be null.
+     * @return true if there are no leases in the lease table
+     *
+     * @throws DependencyException if DynamoDB scan fails in an unexpected way
+     * @throws InvalidStateException if lease table does not exist
+     * @throws ProvisionedThroughputException if DynamoDB scan fails due to lack of capacity
+     */
+    boolean isLeaseTableEmptyForStreamIdentifier(StreamIdentifier streamIdentifier) throws DependencyException,
```

Comment:
LeaseRefresher is a public interface. We need to add interfaces with default to avoid breaking changes.